docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,869 @@
+				<button onclick={this.focusNextChild}>Next</button>
+				<FocusInputChild
+					key={0}
+					focus={this.currentlyFocusedKey === 0 ? this.shouldFocus : undefined}
+					onFocus={() => this.onFocus(0)}
+				/>
+				<FocusInputChild
+					key={1}
+					focus={this.currentlyFocusedKey === 1 ? this.shouldFocus : undefined}
+					onFocus={() => this.onFocus(1)}
+				/>
+				<FocusInputChild
+					key={2}
+					focus={this.currentlyFocusedKey === 2 ? this.shouldFocus : undefined}
+					onFocus={() => this.onFocus(2)}
+				/>
+				<FocusInputChild
+					key={3}
+					focus={this.currentlyFocusedKey === 3 ? this.shouldFocus : undefined}
+					onFocus={() => this.onFocus(3)}
+				/>
+				<FocusInputChild
+					key={4}
+					focus={this.currentlyFocusedKey === 4 ? this.shouldFocus : undefined}
+					onFocus={() => this.onFocus(4)}
+				/>
+			</div>
+		);
+	}
+}
+```
+
+# Managing state
+
+For simple applications where data is not required to flow between many components, state management can be straightforward to deal with. Data can be encapsulated within individual widgets that need it as the most [basic form of state management](#basic-internal-widget-state) within a Dojo application.
+
+As applications grow in complexity and start requiring data to be shared and transferred between multiple widgets, a more robust form of state management is required. Here, Dojo begins to prove its value as a reactive framework, allowing applications to define how data should flow between components, then letting the framework manage change detection and re-rendering. This is done by [wiring widgets and properties together](#intermediate-widget-properties) when declaring VDOM output in a widget's render function.
+
+For large applications, state management can be one of the most challenging aspects to deal with, requiring developers to balance between data consistency, availability and fault tolerance. While a lot of this complexity remains outside the scope of the web application layer, Dojo provides further solutions that help ensure data consistency. The [Dojo Stores](../dojo-stores/supplemental.md) component provides a centralized state store with a consistent API for accessing and managing data from multiple locations within the application.
+
+## Basic: self-encapsulated widget state
+
+Widgets can maintain their own internal state as locally-scoped variables.
+
+Internal state data may directly affect the widget's render output, or may be passed as properties to any child widgets where they in turn directly affect the children's render output. Widgets may also allow their internal state to be changed, for example in response to a user interaction event.
+
+The following example illustrates these patterns:
+
+> src/widgets/MyEncapsulatedStateWidget.tsx
+
+**Function-based variant:**
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+const factory = create();
+
+let myState: string = 'Hello from a stateful widget!';
```

Comment:
@sbinge We can't do this, this would be shared across however many instances of the `MyEncapsulatedStateWidget` existed in the application causing unexpected results.

The most basic form for storing state across renders using function-based widgets is using the `cache` middleware, however it looks like the intent of this example is to re-render when the button is clicked. In this case it would be perfect for the invalidating cache (`icache`) middleware.